### ISSUE TITLE
Examples to cleanup sessions on exit

### ISSUE THREAD

**Issue by mturilli** [radical-cybertools/radical.pilot#1408]
While testing RP I got the following error:
```
2017-08-04 17:23:33,702: radical.pilot       : MainProcess                     : MainThread     : ERROR   : session create failed
Traceback (most recent call last):
  File "/home/rp-user/ve/local/lib/python2.7/site-packages/radical/pilot/session.py", line 154, in __init__
    dburl = dburl)
  File "/home/rp-user/ve/local/lib/python2.7/site-packages/radical/pilot/db/database.py", line 69, in __init__
    raise RuntimeError("Session '%s' already exists." % sid)
RuntimeError: Session 'rp.session.rp-pilot.rp-user.017382.0000' already exists.
Traceback (most recent call last):
  File "./00_getting_started.py", line 38, in <module>
    session = rp.Session()
  File "/home/rp-user/ve/local/lib/python2.7/site-packages/radical/pilot/session.py", line 169, in __init__
    % (dburl, ex))  
radical.pilot.exceptions.PilotException: Couldn't create new session (database URL 'mongodb://two.radical-project.org:27017/rp-user' incorrect?): Session 'rp.session.rp-pilot.rp-user.017382.0000' already exists.
```
I am not sure why this can happen but upon checking, I thought it would make sense for the examples we ship with a master release to clean up after themselves: from 
`session.close(cleanup=False)` to `session.close(cleanup=True)`.

Thoughts?

**Comment by andre-merzky** [comment]
Woah, that is a new one!  I can't immediately see how that could happen.  Can you by any chance verify is an earlier run did indeed use that session ID?  What is strange that the hostname seems to be `rp-pilot`, username `rp-user` - are that the expected values?  Is there any other host/username pair using the same values?

**Comment by mturilli** [comment]
So here what I was able to put together: 
- I run those tests from within our new docker container and rp-pilot was the name I had chosen for the container.
- Run the first test (examples/00_*) all fine. 
- Stopped the container/Started the container
- run the same example and bam! Same sessions name...
- Let docker assign a ramdom string as name (i.e., hostname too), never had a problem
- Choose the same name again, same problem but with a twist: 
  - First time all fine
  - Second time 1 error
  - Third time 2 errors
  - Fourth time 3 errors...

So, basically, it seems that for a reason that escapes my current imagination, the tests start every time from the very first number of the session name.

**Comment by andre-merzky** [comment]
Ah, docker - I see.  Can you please make sure that

 * `$HOME` is set
 * `$HOME/.radical/utils/` survives to the next docket run

That is the place where the ID generator stores the last used ID (see [here](https://github.com/radical-cybertools/radical.utils/blob/master/src/radical/utils/ids.py#L76)).

**Comment by andre-merzky** [comment]
Ah, I just realized that any docker user will, with that image, end up with the same hostname / username pair.  So, yes, cleaning up makes sense - but concurrent users (like in a tutorial setting) would still conflict.

I'll think of something...

**Comment by andre-merzky** [comment]
See https://github.com/radical-cybertools/radical.utils/pull/110 which fixes this now.

**Comment by andre-merzky** [comment]
Matteo - I am going to take take this out of 0.46, as Docker support is not part of it otherwise, ok?

**Comment by andre-merzky** [comment]
This is considered fixed via the above.
